### PR TITLE
(#183) Rules are not canonicalized during matching

### LINKED ISSUE #183: [BUG] Mismatch in `ignore` rule application
While adapting a fork of the plugin to my own use, I discovered a potential inconsistency in the parsing/application of ignore rules, stemming from the fact that when constructing the rule cache, the rules are canonicalized by trimming and lowercasing, but are searched verbatim when the rules are being applied during analysis.

### PR DESCRIPTION
**Pull request by zmeggyesi**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[X] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### What changes did you make? (Give an overview)
Ignore rules in the analyzed code were not being canonicalized during matching, only when constructing the rule cache. This could have led to a potential mis-application of the ignore comments.

closes #183

### CODE REVIEW

**Comment by dkrutskikh** [comment]
@zmeggyesi please update tests for your situations

**Comment by zmeggyesi** [comment]
@dkrutskikh Tests updated with separate case for camel-case rule names

File: lib/src/ignore_info.dart [hunk 1]
```diff
@@ -12,15 +12,18 @@ class IgnoreInfo {
   bool get hasIgnoreInfo =>
       _ignoreMap.isNotEmpty || _ignoreForFileSet.isNotEmpty;
 
-  bool ignoreRule(String ruleId) => _ignoreForFileSet.contains(ruleId);
+  bool ignoreRule(String ruleId) =>
+      _ignoreForFileSet.contains(canonicalize(ruleId));
 
   bool ignoredAt(String ruleId, int line) =>
-      ignoreRule(ruleId) || (_ignoreMap[line]?.contains(ruleId) ?? false);
+      ignoreRule(canonicalize(ruleId)) ||
```
Comment (dkrutskikh) [inline lib/src/ignore_info.dart:19]: @zmeggyesi double canonicalize ruleId

File: lib/src/ignore_info.dart [hunk 2]
```diff
@@ -12,15 +12,18 @@ class IgnoreInfo {
...
+      (_ignoreMap[line]?.contains(canonicalize(ruleId)) ?? false);
+
+  String canonicalize(String ruleId) => ruleId.trim().toLowerCase();
```
Comment (dkrutskikh) [inline lib/src/ignore_info.dart:22]: @zmeggyesi please make `canonicalize` a private function

**Review by dkrutskikh — APPROVED** [review]

**Review by incendial — APPROVED** [review]
@zmeggyesi LGTM, thank you for the contribution.

**Pull request merged**